urlb57eb64056_jacadenac_ISI_Othello_UN_tgz-pJ8-othello_OthelloJ8/src/logica/ia/buscadores/NegaMax.java

```diff
@@ -0,0 +1,84 @@
+package logica.ia.buscadores;
+
+import java.awt.Point;
+import java.util.Set;
+
+import core.EstadoCasilla;
+import core.Tablero;
+import core.Tablero.TipoTablero;
+import logica.ExploradorMovimientos;
+import logica.ia.evaluacion.Evaluacion;
+
+/**
+ * <p>Negamax search is a slightly variant formulation of minimax search
+ * that relies on the zero-sum property of a two-player game.</p>
+ * <p>By definition the value of a position to player A in such a game is
+ * the negation of the value to player B. Thus, the player on move looks
+ * for a move that maximizes the negation of the value of the position
+ * resulting from the move: this successor position must by definition
+ * have been valued by the opponent. </p>
+ * <p>The reasoning of the previous sentence works regardless of whether
+ * A or B is on move. This means that a single computation can be used to
+ * value all positions</p>
+ * <p>This is a coding simplification over minimax, which requires that A
+ * select the move with the maximum-valued successor while B selects the
+ * move with the minimum-valued successor.</p>
+ * 
+ */
+
+public class NegaMax extends BuscadorAbstracto implements /*Buscador,*/ BuscadorSimple {
+
+	@Override
+	public ResultadoBusqueda busquedaSimple(Tablero tablero, TipoTablero tipoTablero,  EstadoCasilla colorJugador, int profundidad, Evaluacion evfuncion) {
+		
+		EstadoCasilla colorOponente;
+		if (colorJugador == EstadoCasilla.BLACK){
+			colorOponente = EstadoCasilla.WHITE;
+		} else {
+			colorOponente = EstadoCasilla.BLACK;
+		}
+		
+		if (profundidad <= 0 || esEstadoFinal(tablero)) {
+			return new ResultadoBusqueda(null, evfuncion.evaluar(tablero, tipoTablero, colorJugador));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem:**
 This line of code lacks validation when processing input data through the following parameter: &#39;evfuncion&#39; (index: 4 | type: Evaluacion). The parameter is exposed to external callers, because its enclosing class and method are publicly accessible. This means that upstream validation, if it exists, can be bypassed. Other validated parameters: &#39;profundidad&#39;, &#39;tablero&#39;, &#39;colorJugador&#39;. Malicious, malformed, or unbounded inputs can cause unexpected runtime behavior or crashes, and can slow performance.

**Fix:**
 Add checks to ensure the validity of the parameter&#39;s value, such as testing it for nullness (for example, using the @NonNull annotation as [described in the Lombok library](https://projectlombok.org/features/NonNull)), emptiness, or equality. Or to prevent direct calls to it, [reduce the method&#39;s visibility](https://docs.oracle.com/javase/tutorial/java/javaOO/accesscontrol.html).

 Learn more about potential threats and guidance from the [Common Weakness Enumeration website](https://cwe.mitre.org/data/definitions/693.html) and the [OWASP Cheat Sheet series](https://cheatsheetseries.owasp.org/cheatsheets/Input_Validation_Cheat_Sheet.html).